locust/load-test/load-test.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+from locust import HttpUser, task, between
+import yaml, json, os
+
+class PuppetDbLoadTest(HttpUser):
+    def response_printer(self, opts, response):
+        if response.status_code == 0:
+            print(response.error)
+            exit(1)
+        elif response.status_code != 200:
+            print(
+            "Request: " + opts['name'] + " \n" +
+            "Method: " + opts['method'] + " \n" +
+            "Response status: " + str(response.status_code) + " \n" +
+            "Response body: " + response.text + " \n" +
+            "-------------------------------------------" + " \n")
+
+    @task
+    def swarm(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        with open(dir_path + '/config.yaml') as stream:
+            config = yaml.safe_load(stream)
+            for opts in config:
+                if (opts['method'] == 'GET'):
+                    limit = opts.get('limit')
```

Comment:
Could we extract the blocks under `if (opts['method'] == 'GET'):` and POST type into separate functions?

Reply:
Yes, I cleaned up the code a bit.